src/lambda/commands/createNewSamApp.ts

```diff
@@ -238,11 +251,33 @@ async function getMainUri(
     }
 }
 
+export async function addInitialLaunchConfiguration(
+    extContext: ExtContext,
+    folder: vscode.WorkspaceFolder,
+    targetUri: vscode.Uri
+): Promise<void> {
+    let configurations = await new SamDebugConfigProvider(extContext).provideDebugConfigurations(folder)
+    if (configurations) {
+        // add configurations that target the new template file
+        let filtered = configurations.filter(
+            config =>
+                isTemplateTargetProperties(config.invokeTarget) &&
+                pathutils.areEqual(
+                    folder.uri.fsPath,
+                    (config.invokeTarget as TemplateTargetProperties).templatePath,
+                    targetUri.fsPath
+                )
+        )
+        await new LaunchConfiguration(folder.uri).addDebugConfigurations(filtered)
+    }
+}
+
 async function addWorkspaceFolder(folder: { uri: vscode.Uri; name?: string }): Promise<void> {
     // No-op if the folder is already in the workspace.
     if (vscode.workspace.getWorkspaceFolder(folder.uri)) {
         return
     }
 
+    // this will never return
```

Comment:
It would return if the folder is being added to a workspace that already has at least one folder open

Reply:
ah misread how this worked, removed